lib/chef/knife/ec2_base.rb

```diff
@@ -207,5 +219,25 @@ def ini_parse(file)
       end
       map
     end
+
+    # All valid platform
+    def valid_platform
+      ["ubuntu", "debian", "centos", "fedora", "rhel", "nginx", "turnkey", "jumpbox", "coreos", "cisco"]
```

Comment:
@btm , Fog's `describe_image` method doesn't return the `platform` attribute for all OS. It returns an image `name` which contains the platform in-between. @piyushawasthi is displaying the platform by checking if the `name` contains any of these platforms.